src/main/java/org/octopusden/releng/versions/NumericVersionFactory.java

```diff
@@ -0,0 +1,53 @@
+package org.octopusden.releng.versions;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class NumericVersionFactory {
+
+    final VersionNames versionNames;
+
+    public NumericVersionFactory(VersionNames versionNames) {
+        this.versionNames = versionNames;
+    }
+
+    public IVersionInfo create(String rawVersion) {
+        Objects.requireNonNull(rawVersion, "version can't be null");
+        final List<String> strings = new ArrayList<>();
+        int l = 0;
+        int r;
+        for (r = 0; r < rawVersion.length(); r++) {
```

Comment:
Is it better to use String.split with regexp?

Reply:
i didn't touch it. Legacy code